src/db/users.rs

```diff
@@ -1,11 +1,48 @@
-use crate::models::user::User;
+use crate::models::user::{Profile, User};
 use crate::schema::users;
 use crypto::scrypt::{scrypt_check, scrypt_simple, ScryptParams};
 use diesel::dsl::{exists, select};
 use diesel::pg::PgConnection;
 use diesel::prelude::*;
 use serde::Deserialize;
 
+/// Return a `Profile` for a given `user` adding the `following` property.
```

Comment:
You're right, I will eliminate that function

Reply:
I have been doing the refactoring, however I have observed that the function to_profile_for is useful, since the `author` field in `Comments` and the `Articles` is of type `Profile` (not `User`), the is_following function is used in association with the `into_profile` method at least six times, and in some cases the code becomes repetitive.

For example, in the case of `find_by_slug` at `db::comments` the code whitout `to_profile_for` would look like this:
```rust
pub fn find_by_slug(conn: &PgConnection, slug: &str, current_user_id: Option<i32>) -> Vec<CommentJson> {
    use crate::db::profiles::is_following;
    let result = comments::table
        .inner_join(articles::table)
        .inner_join(users::table)
        .select((comments::all_columns, users::all_columns))
        .filter(articles::slug.eq(slug))
        .get_results::<(Comment, User)>(conn)
        .expect("Cannot load comments");

    result
        .into_iter()
        .map(|(comment, author)| {
            let following =
            current_user_id.map_or(false, |user_id| is_following(conn, &author, user_id));
            comment.attach(author.into_profile(following))
        }
        )
        .collect()
}
```
Whit `to_profile_for` function this should looks like this
```rust
...
   result
        .into_iter()
        .map(|(comment, author)| comment.attach(to_profile_for(conn, &author, current_user_id))
        .collect()
...
```
Basically I'm relying on the function [toProfileJSONFor](https://github.com/gothinkster/node-express-realworld-example-app/blob/master/models/User.js) of the implementation of the nodeJs backend.

For function `to_profile_for` `Profile` needs a constructor, the other option is convert all `Profile` props to public.

Reply:
Also, as shown in this  [Imgur](https://i.imgur.com/Z3Rwpgg.png)  image, the populate function also needs to be modified to receive an optional argument from the currently loggedin user, otherwise, even if the user is followed, the option to follow will always be shown, "unfollow" option will never be shown. 
I do not know if I'm foolishly insisting on the function `into_profile_for`, I hope to be instructed if there is another option, or the use of the functions `is_following` and `into_profile` together is enough.

Reply:
The only reason why I have not considered implementing `into_profile_for` in the `User` structure is because it would need a `Conn` reference as an argument, and I think a function that needs conn should not go in `models`. I hope to be corrected if I am wrong with that reasoning.

Reply:
I have summarized the implications of updating the type of the `author` property for `ArticleJson` and `CommentJson`, and the implications of the suggested function `into_profile_for` [in the next commit ](https://github.com/luchoman08/realworld-rust-rocket/commit/9fa0efe7f728f24522de31bcdbe30ca928632bbb) belonging to a temporary branch (includes the minor changes of the branch referred to in this pull request, and the recommendations made in previous comments) , if that branch is considered as the best candidate for pull request, I will cancel this pull request and move it to one made from the branch of the link.

Thanks and a thousand apologies for the inconvenience.

Reply:
Makes sense. The only thing which bothers me is the function name.
Maybe something like `user_profile_for(conn, user, user_id)` would be more clear?

Reply:
I checked another branch. It looks good to me and I'm ready to merge it.
If it's possible to split profile changes from clippy to a separate PR it would be really nice.